libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -863,6 +884,14 @@ void AP_GPS::update(void)
 void AP_GPS::update_primary(void)
 {
 #if defined(GPS_BLENDED_INSTANCE)
+    uint8_t primary = _primary.get();
+    if (_primary.get() < 0 || primary >= GPS_MAX_RECEIVERS) {
```

Comment:
why get var again?

Reply:
This was due to the param being a uint and the local variable being an int, I have re-jiged it.